### DISCUSSION TITLE
Filters are not working with PayPal donation, But with PayPal Standard

### DISCUSSION THREAD

**Discussion by alishanvr** [impress-org/givewp#7313] [Developer Q&A]
I am referred here from the priority support team. And I am facing the following issue:

---

**What we want to achieve**

We want that when donation is made then the PayPal title should update and the title should have the information like Give Aid and the country. 
The filters we are using as follows:
* `give_paypal_redirect_args`
* `give_recurring_paypal_args`
* `give_gateway_paypal_redirect_args`

The above filters are working perfect with PayPal Standard but not with the PayPal Donation. 

---

**Code Snippet**

We are using the following filters:

```
add_filter( 'give_paypal_redirect_args', [ $this, 'update_product_title_for_paypal_gateway' ] );
add_filter( 'give_recurring_paypal_args', [ $this, 'update_product_title_for_paypal_gateway' ] );
add_filter( 'give_gateway_paypal_redirect_args', [ $this, 'update_product_title_for_paypal_gateway' ] );


public function update_product_title_for_paypal_gateway( $args ) {

$posted_data = give_clean( filter_input_array( INPUT_POST ) );

// 1 - Check if the form has the gift aid field.
$is_gift_aid = isset( $posted_data['give_gift_aid_accept_term_condition'] ) && $posted_data['give_gift_aid_accept_term_condition'] === 'on' ? 'Yes' : 'No';

if ( strpos( $args['item_name'], '| GiftAid' ) === false ) {
$args['item_name'] .= ' | GiftAid = ' . $is_gift_aid;
}

// 2 - Check if the form has the country field.
$country = isset( $posted_data['select_country'] ) ? $posted_data['select_country'] : '';
if ( ! empty( $country ) ) {

if ( strpos( $args['item_name'], '| Country = ' ) === false ) {
	$args['item_name'] .= ' | Country = ' . $country;
}
}

return $args;
}
```
---

The following filters correctly with PayPal Standard:
* `give_paypal_redirect_args`
* `give_recurring_paypal_args`
* `give_gateway_paypal_redirect_args`

But for the PayPal Donation these filters are not working. 

---

May you please provide us the filters that we should use for the PayPal donation? Or suggest us how to resolve this issue.

**Comment by alishanvr** [comment]
Additionally, 

If we use Stripe and recurring option is enabled then it seems that `give_stripe_create_intent_args` hook not working. As we can not see any metadata to Stripe Dashboard in case of recurring payment. 

But, if it's one off (one time) donation then the metadata is available. 

---

I conducted the through testing and here is the screenshot that may help someone:

![image](https://github.com/impress-org/givewp/assets/5610722/b5fbc808-f1f3-4140-b55f-3781e3466992)

**Text Format** 
[payment-gateway-testing.txt](https://github.com/impress-org/givewp/files/14620837/payment-gateway-testing.txt)

---

Kind Regards,
Ali

**Comment by jonwaldstein** [comment]
Hey @alishanvr 

Thanks for reaching out!

I can confirm you are correct - some hooks you are trying to use do not currently exist for those gateways (at the moment).  To be completely transparent, i'm not sure if the ones you are trying to use worked in the past for those specific gateways.  All of our gateways have been being updated for future version of GiveWP over the last year or so and we have tried our best to preserve hooks along the way.  However, it's certainly possible we missed some or that it was not possible to support them.  

In general my mindset is:
1. If a gateway hook was previously working and is now not working - we should fix that (if possible)
2. If a gateway hook was not previously working but there is enough interest to add - then we should open a feature request

**Comment by alishanvr** [comment]
Hi @jonwaldstein 

Thank you for taking time to answering, and also for confirming that the hooks are not working with the recurring option. 

We are not using these hooks from the past. We just start using it from last month. So, the first question is not relevant (hooks was working in past or not).
And I want to have these hooks working. So, +1 form my side. And also request you to open the feature request. 

If it's on my side (opening the feature request) then please guide me about how I can do this. :)

Kind Regards,

**Reply by jonwaldstein** [reply]
Hey @alishanvr 

For feature requests you can submit here: https://feedback.givewp.com/

Developers are also welcome to open Pull Requests on the GiveWP repositories 😄
